packages/lodestar/src/api/impl/validator/index.ts

```diff
@@ -277,16 +296,43 @@ export function getValidatorApi({chain, config, logger, metrics, network, sync}:
       const startSlot = computeStartSlotAtEpoch(epoch);
       await waitForSlot(startSlot); // Must never request for a future slot > currentSlot
```

Comment:
If we request for current epoch, we need this call. If we request for next epoch, we don't need it. And also the comment was misleading, please fix it

@dadepo how does your test in prater work even with this check?

Reply:
I was a bit confused by this method initially but looking at the implementation, it only waits if the gap between `now` and the next epoch is less than `MAX_API_CLOCK_DISPARITY_MS`, which is set to 1000ms.

Testing on prater seems to work as expected also.